extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,325 @@
+import org.eclipse.smarthome.binding.weatherunderground.internal.json.WeatherUndergroundJsonData;
+import org.eclipse.smarthome.binding.weatherunderground.internal.json.WeatherUndergroundJsonUtils;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link WeatherUndergroundHandler} is responsible for handling the
+ * weather things created to use the Weather Underground Service.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public class WeatherUndergroundHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(WeatherUndergroundHandler.class);
+
+    private static final String URL = "http://api.wunderground.com/api/%APIKEY%/%FEATURES%/%SETTINGS%/q/%QUERY%.json";
+    private static final String FEATURE_CONDITIONS = "conditions";
+    private static final String FEATURE_FORECAST10DAY = "forecast10day";
+    private static final String FEATURE_GEOLOOKUP = "geolookup";
+    private static final Set<String> USUAL_FEATURES = Sets.newHashSet(FEATURE_CONDITIONS, FEATURE_FORECAST10DAY);
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private WeatherUndergroundJsonData weatherData;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private Gson gson;
+
+    public WeatherUndergroundHandler(Thing thing) {
+        super(thing);
+
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing WeatherUnderground handler.");
+        super.initialize();
+
+        WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+        logger.debug("config apikey = {}", config.apikey);
+        logger.debug("config location = {}", config.location);
+        logger.debug("config language = {}", config.language);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            logger.error("Parameter 'apikey' is mandatory and must be configured, disabling thing '{}'",
+                    getThing().getUID());
+            validConfig = false;
```

Comment:
When debugging this stuff myself I found it really useful to do a updateStatus with a text string on errors that says why the config failed.  Then just looking at the page with the binding on it I can see what went wrong, a lot easier for the user too so they don't need to dig through log files to find the issue.